src/Microsoft.SqlTools.ServiceLayer/Migration/Models/StaticTokenCredential.cs

```diff
@@ -0,0 +1,35 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Microsoft.SqlTools.ServiceLayer.Migration.Models
+{
+    /// <summary>
+    /// Temp token creadentials to interact with ArmClient class. 
+    /// The token passed to this class should be a newly request token, because this class doesn't renew the token.
+    /// Once MSAL is rolled out on ADS, we will implement a way to use the same ADS token cache configured by ADS. 
+    /// </summary>
+    internal class StaticTokenCredential : TokenCredential
+    {
+        private readonly AccessToken _token;
+
+        /// <summary>
+        /// Build credentials using a token that will not change.
+        /// </summary>
+        /// <param name="accessToken">Newly created token that should last for the duration of the whole operation.</param>
+        public StaticTokenCredential(string accessToken) => _token = new AccessToken(
+                   accessToken: accessToken,
+                   expiresOn: DateTimeOffset.Now.AddHours(1));  //Default to an hour, the current duration of a newly create token.
```

Comment:
curious, what are the implications of this token expiry? Does this cap the duration of our migrations to 1 hour at a time?

Reply:
That is the expiration time token are getting on token ADS is requesting. A new token is requested on ADS before making this call which should end in less than 5 minutes for around 100 dbs.

Reply:
got it, if those are the types of numbers we're seeing (around 5 mins max) then this shouldn't be an issue. If hitting the 1 hour limit was a possibility I would suggest increasing it but doesn't look like we'll need to